### PR TITLE
664 add context to assessment route

### PR DESCRIPTION
**Pull request by Dhruv-Copods**

### CODE REVIEW

**Comment by richa-copods** [comment]
Fix test cases

File: app/components/tests/TestList.tsx [hunk 1]
```diff
@@ -236,6 +242,10 @@ const TestList = () => {
     location.search,
   ])
 
+  useEffect(() => {
+    setCustomStorage("assessmentSort", sortBy)
```
Comment (richa-copods) [inline app/components/tests/TestList.tsx:246]: Confirm if multiple re-renders are happening or not

**Review by richa-copods — CHANGES_REQUESTED** [review]

**Review by richa-copods — APPROVED** [review]

File: app/context/Common.context.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ interface IStorage {
 }
 
 type CommonContextType = {
-  setCustomStorage: (key: string, value?: Pick<IStorage, "value">) => void
+  setCustomStorage: (key: string, value?: any) => void
```
Comment (anuragP04) [inline app/context/Common.context.tsx:10]: why we have this change here?

File: app/components/tests/TestList.tsx [hunk 3]
```diff
@@ -236,6 +242,10 @@ const TestList = () => {
...
+  }, [sortBy])
```
Comment (anuragP04) [inline app/components/tests/TestList.tsx:247]: try to remove this warning

**Review by anuragP04 — CHANGES_REQUESTED** [review]

File: app/context/Common.context.tsx (hunk 2 above)
Comment (Dhruv-Copods) [reply app/context/Common.context.tsx:10]: as we have discussed there can be multiple type of storage values so giving any specific type here is not possible, thats why I have updated the type

File: app/components/tests/TestList.tsx (hunk 3 above)
Comment (Dhruv-Copods) [reply app/components/tests/TestList.tsx:247]: removing the warning will cause react to rerender multiple times, so if we fix the dependency we will multiple rerenders

Comment (anuragP04) [reply app/components/tests/TestList.tsx:247]: for use effect warning, add lint comment to ignore this one then, but before that check all alternatives

Comment (Dhruv-Copods) [reply app/components/tests/TestList.tsx:247]: sure

**Review by anuragP04 — APPROVED** [review]

**Pull request merged**